docs/source/pages/tutorials/npm_provenance.rst

```diff
@@ -0,0 +1,168 @@
+    component_violates_policy
+    failed_policies
+    passed_policies
+        ['has-verified-provenance']
+
+Additionally, if we had happened to run some more analyses on other versions of ``semver``, we could also apply the policy to them with only a small modification:
+
+.. code-block:: c++
+
+    apply_policy_to("has-verified-provenance", component_id) :-
+        is_component(component_id, purl),
+        match("pkg:npm/semver@.*", purl).
+
+With this modification, all versions of ``semver`` previously analysed will show up when the policy is run again. Like so:
+
+.. code-block:: javascript
+
+    component_satisfies_policy
+        ['1', 'pkg:npm/semver@7.6.2', 'has-verified-provenance']
+        ['2', 'pkg:npm/semver@7.6.0', 'has-verified-provenance']
+    component_violates_policy
+        ['3', 'pkg:npm/semver@1.0.0', 'has-verified-provenance']
+    failed_policies
+        ['has-verified-provenance']
+
+Here we can see that the newer versions, 7.6.2 and 7.6.0, passed the checks, meaning they have verified provenance. The much older version, 1.0.0, did not pass the checks, which is not surprising given that it was published 13 years before this tutorial was made.
+
+However, if we wanted to acknowledge that earlier versions of the artifact do not have provenance, and accept that as part of the policy, we can do that too. For this to succeed we need to extend the policy with more complicated modifications.
+
+.. code-block:: c++
+
+    #include "prelude.dl"
+
+    Policy("has-verified-provenance-or-is-excluded", component_id, "Require a verified provenance file.") :-
+        check_passed(component_id, "mcn_provenance_derived_repo_1"),
+        check_passed(component_id, "mcn_provenance_derived_commit_1"),
+        check_passed(component_id, "mcn_provenance_verified_1"),
+        !exception(component_id).
+
+    Policy("has-verified-provenance-or-is-excluded", component_id, "Make exception for older artifacts.") :-
+        exception(component_id).
+
+    .decl exception(component_id: number)
+    exception(component_id) :-
+        is_component(component_id, purl),
+        match("pkg:npm/semver@[0-6][.].*", purl).
+
+    apply_policy_to("has-verified-provenance-or-is-excluded", component_id) :-
+        is_component(component_id, purl),
+        match("pkg:npm/semver@.*", purl).
+
+In this final policy, we declare (``.decl``) a new rule called ``exception`` that utilises more regular expression in its ``match`` constraint. For this tutorial we have set the exception to accept any versions of ``semver`` that starts with a number between 0 and 6 using the regular expression range component of ``[0-6]``. Then we modify the previous ``Policy`` so that it expects the same three checks to pass, but only if the exception rule is not applicable -- the exclamation mark before the exception negates the requirement. Finally, we add a new ``Policy`` that applies only to those artifacts that match the exception rule.
```

Comment:
```suggestion
In this final policy, we declare (``.decl``) a new rule called ``exception`` that utilises a regular expression in its ``match`` constraint to exclude artifacts that were published before provenance generation was supported. For this tutorial, we have set the exception to accept any versions of ``semver`` that starts with a number between 0 and 6 using the regular expression range component of ``[0-6]``. Then we modify the previous ``Policy`` so that it expects the same three checks to pass, but only if the exception rule is not applicable -- the exclamation mark before the exception negates the requirement. Finally, we add a new ``Policy`` that applies only to those artifacts that match the exception rule.
```